src/coloring/acyclic_coloring.jl

```diff
@@ -0,0 +1,202 @@
+"""
+        color_graph(g::LightGraphs.AbstractGraphs, ::AcyclicColoring)
+
+Returns a coloring vector following the acyclic coloring rules (1) the coloring
+corresponds to a distance-1 coloring, and (2) vertices in every cycle of the
+graph are assigned at least three distinct colors. This variant of coloring is
+called acyclic since every subgraph induced by vertices assigned any two colors
+is a collection of trees—and hence is acyclic.
+
+Reference: Gebremedhin AH, Manne F, Pothen A. **New Acyclic and Star Coloring Algorithms with Application to Computing Hessians**
+"""
+function color_graph(g::LightGraphs.AbstractGraph, ::AcyclicColoring)
+
+    color = zeros(Int, nv(g))
+    forbidden_colors = zeros(Int, nv(g))
+
+    set = DisjointSets{LightGraphs.Edge}([])
+
+    firstVisitToTree = Array{Tuple{Int, Int}, 1}(undef, ne(g))
+    firstNeighbor = Array{Tuple{Int, Int}, 1}(undef, nv(g))
+
+    for v in vertices(g)
+        #enforces the first condition of acyclic coloring
+        for w in outneighbors(g, v)
+            if color[w] != 0
+                forbidden_colors[color[w]] = v
+            end
+        end
+        #enforces the second condition of acyclic coloring
+        for w in outneighbors(g, v)
+            if color[w] != 0 #colored neighbor
+                for x in outneighbors(g, w)
+                    if color[x] != 0 #colored x
+                        if forbidden_colors[color[x]] != v
+                            prevent_cycle(v, w, x, g, color, forbidden_colors, firstVisitToTree, set)
+                        end
+                    end
+                end
+            end
+        end
+
+        color[v] = min_index(forbidden_colors, v)
+
+        #grow star for every edge connecting colored vertices v and w
+        for w in outneighbors(g, v)
+            if color[w] != 0
+                grow_star(v, w, g, firstNeighbor, set, color)
```

Comment:
firstNeighbor: snake case, also, what is this function mutating, v?
The function could have a bang, as: "grow_star!", And the mutated argument should be first

Reply:
It modifies the disjointset set parameter `set`. Fixed.